src/nodes.coffee

```diff
@@ -1947,13 +2064,18 @@ exports.Code = class Code extends Base
 
         haveSplatParam = yes
         if param.splat
-          params.push ref = param.asReference o
-          splatParamName = fragmentsToText ref.compileNode o
+          if param.name instanceof Arr
```

Comment:
Remove these lines, and the test "#4005: `([a = {}]..., b) ->` weirdness" fails. I can’t figure out why.